man/man3/Mail.3
.TH "Mail" 3 "Tue Apr 14 2015" "Version 1.0" "VirtualSCADA" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Mail \- 
.SH SYNOPSIS
.br
.PP
.PP
Inherits \fBMail\fP\&.
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static \fBalwaysFrom\fP ($address, $\fBname\fP=null)"
.br
.ti -1c
.RI "static \fBplain\fP ($view, $\fBdata\fP, $\fBcallback\fP)"
.br
.ti -1c
.RI "static \fBsend\fP ($view, $\fBdata\fP, $\fBcallback\fP)"
.br
.ti -1c
.RI "static \fBqueue\fP ($view, $\fBdata\fP, $\fBcallback\fP, $queue=null)"
.br
.ti -1c
.RI "static \fBqueueOn\fP ($\fBqueue\fP, $view, $\fBdata\fP, $\fBcallback\fP)"
.br
.ti -1c
.RI "static \fBlater\fP ($\fBdelay\fP, $view, $\fBdata\fP, $\fBcallback\fP, $\fBqueue\fP=null)"
.br
.ti -1c
.RI "static \fBlaterOn\fP ($\fBqueue\fP, $\fBdelay\fP, $view, $\fBdata\fP, $\fBcallback\fP)"
.br
.ti -1c
.RI "static \fBhandleQueuedMessage\fP ($job, $\fBdata\fP)"
.br
.ti -1c
.RI "static \fBpretend\fP ($value=true)"
.br
.ti -1c
.RI "static \fBisPretending\fP ()"
.br
.ti -1c
.RI "static \fBgetViewFactory\fP ()"
.br
.ti -1c
.RI "static \fBgetSwiftMailer\fP ()"
.br
.ti -1c
.RI "static \fBfailures\fP ()"
.br
.ti -1c
.RI "static \fBsetSwiftMailer\fP ($swift)"
.br
.ti -1c
.RI "static \fBsetLogger\fP ($logger)"
.br
.ti -1c
.RI "static \fBsetQueue\fP ($\fBqueue\fP)"
.br
.ti -1c
.RI "static \fBsetContainer\fP ($container)"
.br
.in -1c
.SS "Additional Inherited Members"
.SH "Detailed Description"
.PP 
Definition at line 7827 of file _ide_helper\&.php\&.
.SH "Member Function Documentation"
.PP 
.SS "static alwaysFrom ( $address,  $name = \fCnull\fP)\fC [static]\fP"
Set the global from address and name\&.
.PP
\fBParameters:\fP
.RS 4
\fI$address\fP 
.br
\fI$name\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 7837 of file _ide_helper\&.php\&.
.SS "static failures ()\fC [static]\fP"
Get the array of failed recipients\&.
.PP
\fBReturns:\fP
.RS 4
array 
.RE
.PP

.PP
Definition at line 7984 of file _ide_helper\&.php\&.
.SS "static getSwiftMailer ()\fC [static]\fP"
Get the \fBSwift\fP Mailer instance\&.
.PP
\fBReturns:\fP
.RS 4
.RE
.PP

.PP
Definition at line 7974 of file _ide_helper\&.php\&.
.SS "static getViewFactory ()\fC [static]\fP"
Get the view factory instance\&.
.PP
\fBReturns:\fP
.RS 4
.RE
.PP

.PP
Definition at line 7964 of file _ide_helper\&.php\&.
.SS "static handleQueuedMessage ( $job,  $data)\fC [static]\fP"
Handle a queued e-mail message job\&.
.PP
\fBParameters:\fP
.RS 4
\fI$job\fP 
.br
\fI$data\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 7933 of file _ide_helper\&.php\&.
.SS "static isPretending ()\fC [static]\fP"
Check if the mailer is pretending to send messages\&.
.PP
\fBReturns:\fP
.RS 4
bool 
.RE
.PP

.PP
Definition at line 7954 of file _ide_helper\&.php\&.
.SS "static later ( $delay,  $view,  $data,  $callback,  $queue = \fCnull\fP)\fC [static]\fP"
\fBQueue\fP a new e-mail message for sending after (n) seconds\&.
.PP
\fBParameters:\fP
.RS 4
\fI$delay\fP 
.br
\fI$view\fP 
.br
\fI$data\fP 
.br
\fI$callback\fP 
.br
\fI$queue\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed 
.RE
.PP

.PP
Definition at line 7906 of file _ide_helper\&.php\&.
.SS "static laterOn ( $queue,  $delay,  $view,  $data,  $callback)\fC [static]\fP"
\fBQueue\fP a new e-mail message for sending after (n) seconds on the given queue\&.
.PP
\fBParameters:\fP
.RS 4
\fI$queue\fP 
.br
\fI$delay\fP 
.br
\fI$view\fP 
.br
\fI$data\fP 
.br
\fI$callback\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed 
.RE
.PP

.PP
Definition at line 7921 of file _ide_helper\&.php\&.
.SS "static plain ( $view,  $data,  $callback)\fC [static]\fP"
Send a new message when only a plain part\&.
.PP
\fBParameters:\fP
.RS 4
\fI$view\fP 
.br
\fI$data\fP 
.br
\fI$callback\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
int 
.RE
.PP

.PP
Definition at line 7850 of file _ide_helper\&.php\&.
.SS "static pretend ( $value = \fCtrue\fP)\fC [static]\fP"
Tell the mailer to not really send messages\&.
.PP
\fBParameters:\fP
.RS 4
\fI$value\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 7944 of file _ide_helper\&.php\&.
.SS "static queue ( $view,  $data,  $callback,  $queue = \fCnull\fP)\fC [static]\fP"
\fBQueue\fP a new e-mail message for sending\&.
.PP
\fBParameters:\fP
.RS 4
\fI$view\fP 
.br
\fI$data\fP 
.br
\fI$callback\fP 
.br
\fI$queue\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed 
.RE
.PP

.PP
Definition at line 7877 of file _ide_helper\&.php\&.
.SS "static queueOn ( $queue,  $view,  $data,  $callback)\fC [static]\fP"
\fBQueue\fP a new e-mail message for sending on the given queue\&.
.PP
\fBParameters:\fP
.RS 4
\fI$queue\fP 
.br
\fI$view\fP 
.br
\fI$data\fP 
.br
\fI$callback\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
mixed 
.RE
.PP

.PP
Definition at line 7891 of file _ide_helper\&.php\&.
.SS "static send ( $view,  $data,  $callback)\fC [static]\fP"
Send a new message using a view\&.
.PP
\fBParameters:\fP
.RS 4
\fI$view\fP 
.br
\fI$data\fP 
.br
\fI$callback\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 7863 of file _ide_helper\&.php\&.
.SS "static setContainer ( $container)\fC [static]\fP"
Set the IoC container instance\&.
.PP
\fBParameters:\fP
.RS 4
\fI$container\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 8028 of file _ide_helper\&.php\&.
.SS "static setLogger ( $logger)\fC [static]\fP"
Set the log writer instance\&.
.PP
\fBParameters:\fP
.RS 4
\fI$logger\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
$this 
.RE
.PP

.PP
Definition at line 8006 of file _ide_helper\&.php\&.
.SS "static setQueue ( $queue)\fC [static]\fP"
Set the queue manager instance\&.
.PP
\fBParameters:\fP
.RS 4
\fI$queue\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
$this 
.RE
.PP

.PP
Definition at line 8017 of file _ide_helper\&.php\&.
.SS "static setSwiftMailer ( $swift)\fC [static]\fP"
Set the \fBSwift\fP Mailer instance\&.
.PP
\fBParameters:\fP
.RS 4
\fI$swift\fP 
.RE
.PP
\fBReturns:\fP
.RS 4
void 
.RE
.PP

.PP
Definition at line 7995 of file _ide_helper\&.php\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for VirtualSCADA from the source code\&.


virtualscada_laravel/config/mail.php
<?php

return [

	/*
	|--------------------------------------------------------------------------
	| Mail Driver
	|--------------------------------------------------------------------------
	|
	| Laravel supports both SMTP and PHP's "mail" function as drivers for the
	| sending of e-mail. You may specify which one you're using throughout
	| your application here. By default, Laravel is setup for SMTP mail.
	|
	| Supported: "smtp", "mail", "sendmail", "mailgun", "mandrill", "log"
	|
	*/

	'driver' => env('MAIL_DRIVER', 'smtp'),

	/*
	|--------------------------------------------------------------------------
	| SMTP Host Address
	|--------------------------------------------------------------------------
	|
	| Here you may provide the host address of the SMTP server used by your
	| applications. A default option is provided that is compatible with
	| the Mailgun mail service which will provide reliable deliveries.
	|
	*/

	'host' => env('MAIL_HOST', 'smtp.mailgun.org'),

	/*
	|--------------------------------------------------------------------------
	| SMTP Host Port
	|--------------------------------------------------------------------------
	|
	| This is the SMTP port used by your application to deliver e-mails to
	| users of the application. Like the host we have set this value to
	| stay compatible with the Mailgun e-mail application by default.
	|
	*/

	'port' => env('MAIL_PORT', 587),

	/*
	|--------------------------------------------------------------------------
	| Global "From" Address
	|--------------------------------------------------------------------------
	|
	| You may wish for all e-mails sent by your application to be sent from
	| the same address. Here, you may specify a name and address that is
	| used globally for all e-mails that are sent by your application.
	|
	*/

	'from' => ['address' => 'smu.virtual.scada@gmail.com', 'name' => 'SCADA webmaster'],

	/*
	|--------------------------------------------------------------------------
	| E-Mail Encryption Protocol
	|--------------------------------------------------------------------------
	|
	| Here you may specify the encryption protocol that should be used when
	| the application send e-mail messages. A sensible default using the
	| transport layer security protocol should provide great security.
	|
	*/

	'encryption' => 'ssl',

	/*
	|--------------------------------------------------------------------------
	| SMTP Server Username
	|--------------------------------------------------------------------------
	|
	| If your SMTP server requires a username for authentication, you should
	| set it here. This will get used to authenticate with your server on
	| connection. You may also set the "password" value below this one.
	|
	*/

	'username' => env('MAIL_USERNAME'),

	/*
	|--------------------------------------------------------------------------
	| SMTP Server Password
	|--------------------------------------------------------------------------
	|
	| Here you may set the password required by your SMTP server to send out
	| messages from your application. This will be given to the server on
	| connection so that the application will be able to send messages.
	|
	*/

	'password' => env('MAIL_PASSWORD'),

	/*
	|--------------------------------------------------------------------------
	| Sendmail System Path
	|--------------------------------------------------------------------------
	|
	| When using the "sendmail" driver to send e-mails, we will need to know
	| the path to where Sendmail lives on this server. A default path has
	| been provided here, which will work well on most of your systems.
	|
	*/

	'sendmail' => '/usr/sbin/sendmail -bs',

	/*
	|--------------------------------------------------------------------------
	| Mail "Pretend"
	|--------------------------------------------------------------------------
	|
	| When this option is enabled, e-mail will not actually be sent over the
	| web and will instead be written to your application's logs files so
	| you may inspect the message. This is great for local development.
	|
	*/

	'pretend' => false,

];
